tests/functional_test.go

```diff
@@ -103,235 +102,6 @@ condition conds(s: string) {
 	require.Nil(t, checkResp)
 }
 
-// TODO make a unit test from this
-func TestCheckWithQueryCacheEnabled(t *testing.T) {
```

Comment:
I never liked this test introduced in https://github.com/openfga/openfga/pull/1059. The bug was in `check.go` but we wrote it as a functional tests. So I rewrote it in two parts.